app/uk/gov/hmrc/agentclientrelationships/connectors/DesConnector.scala

```diff
@@ -113,20 +111,44 @@ class DesConnector @Inject() (httpClient: HttpClient, metrics: Metrics, agentCac
     }
   }
 
-  def desHeaders(authToken: String, env: String)(implicit hc: HeaderCarrier): Seq[(String, String)] =
-    Seq(
-      HeaderNames.authorisation -> s"Bearer $authToken",
-      Environment               -> env,
-      CorrelationId             -> UUID.randomUUID().toString
-    ) ++ hc.sessionId.fold(Seq.empty[(String, String)])(x => Seq(SessionId -> x.value)) ++
-      hc.requestId.fold(Seq(RequestId -> UUID.randomUUID().toString))(x => Seq(RequestId -> x.value))
+  /*
+   * If the service being called is external (e.g. DES/IF in QA or Prod):
+   * headers from HeaderCarrier are removed (except user-agent header).
+   * Therefore, required headers must be explicitly set.
+   * See https://github.com/hmrc/http-verbs?tab=readme-ov-file#propagation-of-headers
+   * */
+
+  def desHeaders(authToken: String, env: String, isInternalHost: Boolean)(implicit
+    hc: HeaderCarrier
+  ): Seq[(String, String)] = {
+
+    val additionalHeaders =
+      if (isInternalHost) Seq.empty
+      else
+        Seq(
+          HeaderNames.authorisation -> s"Bearer $authToken",
+          HeaderNames.xSessionId    -> hc.sessionId.map(_.value).getOrElse("sessionId not available"),
```

Comment:
I think we should not send sessionId when not available

Reply:
I have changed it to not include the header when sessionId is not available